collectors/logsmanagement.plugin/plugin_logsmanagement.c

```diff
@@ -0,0 +1,411 @@
+        stats_chart_data->dim_circ_buff_mem_total_arr[i] = 
+            rrddim_add( stats_chart_data->st_circ_buff_mem_total, 
+                        p_file_info->chart_name, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+        stats_chart_data->dim_circ_buff_mem_uncompressed_arr[i] = 
+            rrddim_add( stats_chart_data->st_circ_buff_mem_uncompressed, 
+                        p_file_info->chart_name, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+        stats_chart_data->dim_circ_buff_mem_compressed_arr[i] = 
+            rrddim_add( stats_chart_data->st_circ_buff_mem_compressed, 
+                        p_file_info->chart_name, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+
+        /* Compression stats - add dimensions */
+        stats_chart_data->dim_compression_ratio[i] = 
+            rrddim_add( stats_chart_data->st_compression_ratio, 
+                        p_file_info->chart_name, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+
+        /* DB disk usage stats - add dimensions */
+        stats_chart_data->dim_disk_usage[i] = 
+            rrddim_add( stats_chart_data->st_disk_usage, 
+                        p_file_info->chart_name, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+
+        chart_data_arr[i] = callocz(1, sizeof(struct Chart_meta));
+        memcpy(chart_data_arr[i], &chart_types[p_file_info->log_type], sizeof(struct Chart_meta));
+        chart_data_arr[i]->init(p_file_info, chart_data_arr[i]);
+
+        /* Custom charts - initialise */
+        for(int cus_off = 0; p_file_info->parser_cus_config[cus_off]; cus_off++){
+            chart_data_arr[i]->chart_data_cus_arr = reallocz( chart_data_arr[i]->chart_data_cus_arr, 
+                                                              (cus_off + 1) * sizeof(Chart_data_cus_t *));
+            chart_data_arr[i]->chart_data_cus_arr[cus_off] = callocz(1, sizeof(Chart_data_cus_t));
+
+            RRDSET *st_cus = rrdset_find_active_bytype_localhost( p_file_info->chart_name, 
+                                                                  p_file_info->parser_cus_config[cus_off]->chart_name);
+            if(st_cus) chart_data_arr[i]->chart_data_cus_arr[cus_off]->st_cus = st_cus;
+            else {
+                chart_data_arr[i]->chart_data_cus_arr[cus_off]->st_cus = rrdset_create_localhost(
+                        (char *) p_file_info->chart_name
+                        , p_file_info->parser_cus_config[cus_off]->chart_name
+                        , NULL
+                        , "regex charts"
+                        , NULL
+                        , p_file_info->parser_cus_config[cus_off]->chart_name
+                        , "matches/s"
+                        , "logsmanagement.plugin"
+                        , NULL
+                        , NETDATA_CHART_PRIO_CUS + cus_off
+                        , p_file_info->update_every
+                        , RRDSET_TYPE_AREA
+                );
+                // rrdset_next() and rrdset_done() need to be run for this
```

Comment:
I agree, please, add them. :)

`cgroup` has good examples, like [this](https://github.com/netdata/netdata/blob/185f5a183b423c4fb76bbcf70778f66e69c31fc1/collectors/cgroups.plugin/sys_fs_cgroup.c#L2851-L2873)

Reply:
I am still starting the review, so probably either I am missing what is coming or you need to remove this comment.

Reply:
Hmmm, it's not very clear from my comment, but basically what I mean is that you will see later in the code that I have this:
```
            if(chart_data_arr[i]->chart_data_cus_arr[cus_off]->need_rrdset_done){
                rrdset_done(chart_data_arr[i]->chart_data_cus_arr[cus_off]->st_cus);
            }
```

So `need_rrdset_done` is effectively a `boolean` : when it is 1,  `rrdset_next()` and `rrdset_done()` are executed for this chart and when it is 0, they have been executed in a previous iteration. It is needed due to the way I have implemented custom chart dimension grouping - I don't think there is anything to be done here.